src/shared/utils/guard.ts

```diff
@@ -0,0 +1,43 @@
+import { SchemaId, SignedWrappedDocument } from "../..";
+import {
+  OpenAttestationDocument as OpenAttestationDocumentV3,
+  VerifiableCredentialProofBase,
+  VerifiableCredentialProofSigned,
+  WrappedDocument as WrappedDocumentV3
+} from "../../3.0/types";
+import {
+  ArrayProof,
+  OpenAttestationDocument as OpenAttestationDocumentV2,
+  Signature,
+  WrappedDocument as WrappedDocumentV2
+} from "../../2.0/types";
+import { getSchema, validateSchema as validate } from "../validate";
+
+export const isWrappedV3Document = (document: any): document is WrappedDocumentV3<OpenAttestationDocumentV3> => {
+  if (!document || typeof document !== "object" || document.version !== SchemaId.v3) return false;
```

Comment:
wondering why you don't check this with runtypes

Reply:
we already have types generated from json-schema, so I just check with this.